### PR TITLE
refactor(Button): enable loading state for button

### PR DESCRIPTION
**Pull request by araujobarret**
### Description

Fixes loading states similar to what was done in #243. 
Co-authored by @gkartalis 

https://github.com/user-attachments/assets/40794982-fd80-449f-8a96-517e172cf934

### CODE REVIEW

File: src/elements/Button/Button.tsx [hunk 1]
```diff
@@ -213,46 +221,42 @@ export const Button = ({
...
+              {iconPosition === "right" && !!icon && (
                 <>
                   <Spacer x={0.5} />
                   {icon}
                 </>
-              ) : null}
+              )}
 
-              {loading ? (
+              {loading && (
```
Comment (gkartalis) [inline src/elements/Button/Button.tsx:249]: 🤔 can't believe that this was the issue 🤔

**Review by gkartalis — APPROVED** [review]

File: src/elements/Button/Button.tsx (hunk 1 above)
Comment (araujobarret) [reply src/elements/Button/Button.tsx:249]: it wasn't we we missing the loader colors and the transparencies to the text too

Comment (gkartalis) [reply src/elements/Button/Button.tsx:249]: I get that the colors were missing but what about the size of the button? It is not affected by the colors right?

**Comment by araujobarret** [comment]
Added a missing fix on the follow button here https://github.com/artsy/palette-mobile/pull/253/commits/1929b176415957b76225f13fcc4c32fa78316f59

result:

https://github.com/user-attachments/assets/21c308b9-6377-459a-b443-4c5182aabf12

**Review by gkartalis — APPROVED** [review]

**Comment by artsyit** [comment]
:rocket: PR was released in [`v13.2.30`](https://github.com/artsy/palette-mobile/releases/tag/v13.2.30) :rocket:

**Pull request merged**